samcli/local/docker/container.py

```diff
@@ -225,6 +226,45 @@ def create(self):
 
         return self.id
 
+    def _create_mapped_symlink_files(self, mount_mode: str) -> Dict[str, Dict[str, str]]:
+        """
+        Resolves any top level symlinked files and folders that are found on the
+        host directory and creates additional bind mounts to correctly map them
+        inside of the container.
+
+        Parameters
+        ----------
+        mount_mode: str
+            The mode to mount the file or directory
+
+        Returns
+        -------
+        Dict[str, Dict[str, str]]
+            A dictonary representing the resolved file or directory and the bound path
+            on the container
+        """
+        additional_volumes = {}
+
+        with os.scandir(self._host_dir) as directory_iterator:
+            for file in directory_iterator:
+                if not file.is_symlink():
+                    continue
+
+                host_resolved_path = os.path.realpath(file.path)
+                container_full_path = pathlib.Path(self._working_dir, file.name).as_posix()
+
+                additional_volumes[host_resolved_path] = {
+                    "bind": container_full_path,
+                    "mode": mount_mode,
+                }
```

Comment:
As we discussed offline, we need to check with security about this part before releasing it.